Source/Core/DolphinQt2/Config/CheatWarningWidget.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+#include <QTimer>
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "DolphinQt2/Settings.h"
+
+CheatWarningWidget::CheatWarningWidget(const std::string& game_id) : m_game_id(game_id)
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  setHidden(true);
+
+  connect(&Settings::Instance(), &Settings::EnableCheatsChanged, this, &CheatWarningWidget::Update);
+  connect(this, &CheatWarningWidget::EmulationStarted, [this] { Update(true); });
+  connect(this, &CheatWarningWidget::EmulationStopped, [this] { Update(false); });
+
+  Update(Core::IsRunning());
+}
+
+void CheatWarningWidget::CreateWidgets()
+{
+  auto* icon = new QLabel;
+
+  QPixmap warning_icon = style()->standardIcon(QStyle::SP_MessageBoxWarning).pixmap(24, 24);
```

Comment:
Hmm, I wonder how this pixmap size works across different DPIs. It'd probably be better if we used something based off of Qt's metrics—maybe `1.5 * QFontMetrics(font()).height()`, or something like that?